common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -2097,7 +2098,12 @@ def _update_item_from_fields(self, user_id, course_key, block_key, partitioned_f
 
                 # fetch and return the new item--fetching is unnecessary but a good qc step
                 new_locator = course_key.make_usage_key(block_key.type, block_key.id)
-                return self.get_item(new_locator, **kwargs)
+                new_item = self.get_item(new_locator, **kwargs)
+
+                if block_key.type == 'library_content':
```

Comment:
@cpennington Understood regarding it is not the appropriate place to do this.
I investigated more [LibraryContentModule](https://github.com/edx/edx-platform/blob/b5ab64505af8d83dfe147771d799af5741056024/common/lib/xmodule/xmodule/library_content_module.py#L126) and [LibraryContentDescriptor](https://github.com/edx/edx-platform/blob/b5ab64505af8d83dfe147771d799af5741056024/common/lib/xmodule/xmodule/library_content_module.py#L366).

During import only [definition_from_xml](https://github.com/edx/edx-platform/blob/b5ab64505af8d83dfe147771d799af5741056024/common/lib/xmodule/xmodule/library_content_module.py#L623) method of `LibraryContentDescriptor` class is called that imports the *Library Content Module*.

I am still unsure when and what is the right place/method for calling [update_children](https://github.com/edx/edx-platform/blob/b5ab64505af8d83dfe147771d799af5741056024/common/lib/xmodule/xmodule/library_tools.py#L127) of `LibraryToolsService` class.
[LibraryToolsService](https://github.com/edx/edx-platform/blob/b5ab64505af8d83dfe147771d799af5741056024/common/lib/xmodule/xmodule/library_tools.py#L18)

Please advise, Thanks.